repaygithub/ui-tools
# CACTUS-508 :: ESLint Parser Change and Jest Sonar Swap

I updated `lodash`, but the real change of this PR is in the eslint config.  `babel-parser` has been archived, so this PR swaps it out for the now-supported `@babel/eslint-parser`.  This is a breaking change, because `@babel/core` is now a peer dependency, and consumers must have a Babel config file on their project.

I also replaced `jest-sonar-reporter` with `jest-sonar`, as I did for Cactus

### Testing

1. Pull down this branch and `yarn install`
2. Run `yarn lint`, and make sure there are no problems.
3. `cd modules/eslint-config` and `yalc publish`
4. `cd` over to Cactus and run `yalc add @repay/eslint-config`.
5. Open up `package.json`.  Delete `@repay/eslint-config` from `devDependencies` (it's now linked by yalc) and add `"@babel/core": "^7.12.16"`.
6. Run `yarn install`
7. Open up `babel.config.js` inside the Cactus root.  Add the following line before the return statement: `api.cache.never()` (you'll need to add `api` to the function args).  Babel will complain if you don't do this (I assume we haven't run into this before because most usages of Babel inside this repo probably defer to the module-level `.babelrc` files instead of this global config).
8. Run `yarn lint`, and make sure there are no problems.
9. Go back to `ui-tools`, then run `yarn w @repay/testing-tools test`, and make sure that a `test-report.xml` fils is properly generated.
